transport/tchannel/inbound.go

```diff
@@ -0,0 +1,73 @@
+package tchannel
+
+import (
+	"go.uber.org/yarpc/api/transport"
+	"go.uber.org/yarpc/internal/introspection"
+	"go.uber.org/yarpc/internal/sync"
+)
+
+// Inbound receives YARPC requests over TChannel. It may be constructed using
+// the NewInbound method on a tchannel.Transport.
+type Inbound struct {
+	once      sync.LifecycleOnce
+	transport *Transport
+}
+
+// NewInbound returns a new TChannel inbound backed by a shared TChannel
+// transport.
+// There should only be one inbound for TChannel since all outbounds send the
+// listening port over non-ephemeral connections so a service can deduplicate
```

Comment:
If we do it, we should do it now. I’ll have to rope in @prashantv to assess whether that is feasible. It might mean having a unique channel per *inbound* and affiliating all outbounds with the first. I also need to ask @prashantv how one configures TChannel to expressly use ephemeral connections on the outbound. We should have a transport.Ephemeral option that NewOutbound would accept (in its absence, we would init with the host:port of the *first* inbound)